core/src/main/java/com/linecorp/armeria/internal/server/annotation/KotlinUtil.java

```diff
@@ -0,0 +1,172 @@
+            isReturnTypeUnit = kotlinUtilClass.getMethod("isReturnTypeUnit", Method.class);
+        } catch (ClassNotFoundException | NoSuchMethodException e) {
+            // ignore
+        } finally {
+            IS_CONTINUATION = isContinuation;
+            IS_SUSPENDING_FUNCTION = isSuspendingFunction;
+            IS_RETURN_TYPE_UNIT = isReturnTypeUnit;
+        }
+
+        boolean isKotlinReflectionPresent = false;
+        try {
+            getClass("kotlin.reflect.full.KClasses");
+            isKotlinReflectionPresent = true;
+        } catch (ClassNotFoundException e) {
+            // ignore
+        } finally {
+            IS_KOTLIN_REFLECTION_PRESENT = isKotlinReflectionPresent;
+        }
+
+        Class<? extends Annotation> metadataClass = null;
+        try {
+            metadataClass = (Class<? extends Annotation>) getClass("kotlin.Metadata");
+        } catch (ClassNotFoundException e) {
+            // ignore
+        } finally {
+            METADATA_CLASS = metadataClass;
+        }
+    }
+
+    /**
+     * Returns a method which invokes Kotlin suspending functions.
+     */
+    @Nullable
+    static MethodHandle getCallKotlinSuspendingMethod() {
+        return CALL_KOTLIN_SUSPENDING_METHOD;
+    }
+
+    /**
+     * Returns true if a method is written in Kotlin.
+     */
+    static boolean isKotlinMethod(Method method) {
+        return METADATA_CLASS != null &&
+               method.getDeclaringClass().getAnnotation(METADATA_CLASS) != null;
+    }
+
+    /**
+     * Returns true if a method is a suspending function.
+     */
+    static boolean isSuspendingFunction(Method method) {
+        try {
+            return isKotlinMethod(method) &&
+                   IS_KOTLIN_REFLECTION_PRESENT &&
+                   IS_SUSPENDING_FUNCTION != null &&
+                   (Boolean) IS_SUSPENDING_FUNCTION.invoke(null, method);
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    /**
+     * Returns true if a class is kotlin.coroutines.Continuation.
+     */
+    static boolean isContinuation(Class<?> type) {
+        try {
+            return IS_CONTINUATION != null &&
+                   (boolean) IS_CONTINUATION.invoke(null, type);
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    /**
+     * Returns true if a method is suspending function and it returns kotlin.Unit.
+     */
+    static boolean isSuspendingAndReturnTypeUnit(Method method) {
+        try {
+            return isSuspendingFunction(method) &&
+                   IS_RETURN_TYPE_UNIT != null &&
+                   (boolean) IS_RETURN_TYPE_UNIT.invoke(null, method);
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    private static Class<?> getClass(String name) throws ClassNotFoundException {
+        return Class.forName(name, true, KotlinUtil.class.getClassLoader());
+    }
+
+    private KotlinUtil() {
+    }
```

Comment:
nit: Merge two lines?
```suggestion
    private KotlinUtil() {}
```